### PR TITLE
Add initial test for `Client.onmessage()`

### PR DESCRIPTION
**Pull request by portante**
This is layered on top of PR #12.

### CODE REVIEW

**Comment by portante** [comment]
This code does not work because the POST request does not complete synchronously with the return of `onmessage()`.  We need a way to either wait for that event or mock out the `post()` method to return a mocked option for `RequestBase`.

**Review by webbnh — CHANGES_REQUESTED** [review]
I found [this reference](https://jestjs.io/docs/asynchronous) to be illustrative.

Ordinarily, you don't need to use [the `async` keyword](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) unless you are using `await`; however, the code here doesn't use `await`, and so the `async` declaration at line 49 doesn't accomplish anything.  And, according to the reference above, since this function doesn't return a `Promise`, Jest doesn't know to wait for anything, and so it doesn't work.

(Let me know if you want to chat about this; I probably don't know more about it than you do, but I'm happy to kibitz.... 😀)

**Comment by portante** [comment]
This might be useful: https://www.npmjs.com/package/flush-promises

**Comment by webbnh** [comment]
The [source for `onmessage()`](https://github.com/distributed-system-analysis/smee-client/blob/2fc2414d1182557543d8c54f8aa0378800018cc8/src/client.ts#L41) looks like this:
```typescript
  onmessage (msg: any): void {

  ...

    req.send(body)
      .then((res) => {
        this.logger.info(`SUCCESS: ${req.method} ${req.url} - ${res.status}`)
      })
      .catch((err) => {
        this.logger.error(`ERROR: ${req.method} ${req.url}`, err)
      })
  }
```
The result of `catch()` is [a promise equivalent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch) to the one returned by `send()`, which could be used to determine when the request is complete, but for the fact that `onmessage()` is lacking a `return` statement and declared to be `void`.

So, I think all you need to do to make this testable is remove the `void` and add a `return`.

File: src/client.test.ts [hunk 1]
```diff
@@ -46,6 +47,30 @@ describe('Client', () => {
...
+
+    // Message that will be "received".
+    const msg = { data: `{"query": {"foo": "bar", "then": "now"}, "host": "${srcHost}", "body": "payload", "key1": "one", "key2": "two"}` }
+
+    // "Receive" the message.
+    client.onmessage(msg)
+
+    // Ensure all POST operations complete before test ends.
+    await flushPromises()
+    await flushPromises()
```
Comment (webbnh) [inline src/client.test.ts:69]: We have to call it twice??

**Review by webbnh — APPROVED** [review]

File: src/client.test.ts (hunk 1 above)
Comment (portante) [reply src/client.test.ts:69]: Apparently.  I am not going to dig into this further.  This satisfied my curiosity about `flush-promises` and so I am considering this experiment "closed" as `gosmee` is our new "go-to" smee server/client.

**Pull request merged**